oak_functions/wasm/src/lib.rs

```diff
@@ -30,6 +30,7 @@ use hashbrown::HashMap;
 use oak_functions_abi::proto::{ExtensionHandle, OakStatus, Request, Response, StatusCode};
 use oak_functions_extension::{ExtensionFactory, OakApiNativeExtension};
 use oak_logger::{Level, OakLogger};
+use oak_remote_attestation::crypto::get_sha256;
```

Comment:
@conradgrobler I am not sure, that adding this dependency here is a good idea. Do you agree?

Reply:
I don't think a reference to the remote attestation create should be added to this crate. The runtime should generate the hash before creating the wasm instance rather than expecting the wasm crate to generate a hash.

Reply:
`ring` cannot be used in bare-metal environments, and rust crypto `sha2` cannot be used in uefi apps , which is why we have the two implementations.

I am not sure the reason for moving the hash into the wasm handler, as it does not seem like something it should be responsible for. Can you give a bit more context for this refactoring?

Reply:
Ah, thank you, Conrad! 

re: refactoring: I am trying to simplify the way we configure/pass around the configuration around the Oak Functions Runtime. 

I thought we could easily compute the `wasm_hash` from the bytes within the WasmHandler because giving it as an argument is just an additional point for inconsistency.

But thanks for reminding me that the uefi/bare-metal make that impossible, so I'll give it as argument to WasmHandler.

Reply:
I don't think adding the `wasm_hash` as argument to WasmHandler as actually improves the code, so I am abandoning this PR.

I still think it would be nice if the WasmHandler stores some introspective about the running Wasm Code even after it was initially created (hence storing the `wasm_hash`), but there is no clear reason for it at the moment.